use_case_examples/hybrid_model/infer_hybrid_llm_generate.py

```diff
@@ -0,0 +1,89 @@
+"""Showcase for the hybrid model converter."""
+import time
+from pathlib import Path
+
+import torch
+from torch.backends import mps
+from transformers import AutoModelForCausalLM, AutoTokenizer, PreTrainedTokenizer, TextStreamer
+
+from concrete.ml.torch.hybrid_model import FHEMode, HybridFHEModel
+
+if __name__ == "__main__":
+    configs = [
+        ("transformer.h.0.mlp", 0.934),
+        (["transformer.h.0.mlp", "transformer.h.1.mlp"], 0.42),
+        ("transformer.h.0.mlp.c_proj", 0.986),
+        ("transformer.h.0.attn.c_proj", 0.986),
+    ]
+    config_index = 3
+    config = configs[config_index][0]
+
+    device = "cpu"
+    if torch.cuda.is_available():
+        device = "cuda"
+    if mps.is_available():
+        device = "mps"
+    print(f"Using device: {device}")
+
+    # Get GPT2 from Huggingface
+    # TODO: migrate to auto-model with model_name
+    model_name = "gpt2"
+    model_name_no_special_char = model_name.replace("/", "_")
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+    model = AutoModelForCausalLM.from_pretrained(
+        model_name,
+        device_map=device,
+        trust_remote_code=True,
+    )
+
+    # Modify model to use remote FHE server instead of local weights
+    hybrid_model = HybridFHEModel(
+        model,
+        config,
+        server_remote_address="http://0.0.0.0:8000",
+        model_name=f"{model_name}_{config_index}",
+        verbose=False,
+    )
+    path_to_clients = Path(__file__).parent / "clients"
+    hybrid_model.init_client(path_to_clients=path_to_clients)
+    for module in hybrid_model.remote_modules.values():
+        module.fhe_local_mode = FHEMode.REMOTE
```

Comment:
Would be great to add a comment here. Also I guess we could have a function that does this in hybrid model instead of letting users do this loop

Reply:
This is actually redundant with what is done in the forward of the HybridModel.

BUT to have all remote module calling the server when using the Module directly and not the hybrid model itself this is needed.

Would you prefer to have a setter for this attribute for the hybrid model that also sets this attribute in the remote modules?